### PR TITLE
Feature/data in models

### PR DESCRIPTION
**Pull request by Leboroz**
# Learning objectives

- Set up associations between models.
- Write SQL queries with Active Record.

## Description

> Now it's time to set up our models. To start with, we have tables for Users, Posts, Comments, and Likes, which means we need a model for each one. You've already set the foreign key in the table schema but be sure to write it here too!

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+class User < ApplicationRecord
+  has_many :comments
+  has_many :posts
+  has_many :likes
```
Comment (mikethreels) [inline app/models/user.rb]: - [ ] you need to set the foreign key as well for the association

File: app/models/user.rb [hunk 2]
```diff
@@ -0,0 +1,9 @@
+class User < ApplicationRecord
+  has_many :comments
+  has_many :posts
+  has_many :likes
+
+  def most_recent_posts
+    posts[0, 3]
+  end
```
Comment (mikethreels) [inline app/models/user.rb]: - [ ] to be sure you're getting the correct posts you should order them first, which will look something like the below.
```
posts.order('created_at Desc').limit(3)
```

File: app/models/post.rb [hunk 3]
```diff
@@ -0,0 +1,13 @@
+class Post < ApplicationRecord
+  belongs_to :user
+  has_many :comments
+  has_many :likes
+
+  def increase_user_counter
+    User.find_by(id: user.id).update(post_counter: user.post_counter.nil? ? 1 : ++user.post_counter)
```
Comment (mikethreels) [inline app/models/post.rb]: - [ ] if you set the foreign key in the `User` model and set the belongs to in this model to the foreign and class_name: user the following can look like below ```  author.increment!(:posts_counter)```
- [ ] [OPTIONAL]after that you can also add something like `after_save :update_post_counter` and make this method private

File: app/models/post.rb [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+  has_many :comments
+  has_many :likes
+
+  def increase_user_counter
+    User.find_by(id: user.id).update(post_counter: user.post_counter.nil? ? 1 : ++user.post_counter)
+  end
+
+  def most_recent_comments
+    comments[0, 5]
+  end
```
Comment (mikethreels) [inline app/models/post.rb]: - [ ] to be sure you're getting the correct comments you should sort them first, which will look something like the below.
```
def recent_comments
  comments.order('created_at Desc').limit(5)
end

```

File: app/models/comment.rb [hunk 5]
```diff
@@ -0,0 +1,8 @@
+class Comment < ApplicationRecord
+  belongs_to :user
+  belongs_to :post
+
+  def increase_comments_counter
+    Post.find_by(id: post.id).update(commentsCounter: post.commentsCounter.nil? ? 1 : ++post.commentsCounter)
+  end
```
Comment (mikethreels) [inline app/models/comment.rb]: - [ ] in this model you can do the same as I suggested in the user model

File: app/models/like.rb [hunk 6]
```diff
@@ -0,0 +1,8 @@
+class Like < ApplicationRecord
+  belongs_to :user
+  belongs_to :post
+
+  def increase_like_counter
+    Post.find_by(id: post.id).update(likesCounter: post.likesCounter.nil? ? 1 : ++post.likesCounter)
+  end
```
Comment (mikethreels) [inline app/models/like.rb]: - [ ] in this model you can do the same as I suggested in the user model

**Review by mikethreels — CHANGES_REQUESTED** [review]
Hi Leo, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Deepakdanger — APPROVED** [review]
Hi @Leboroz, 

### Status: Approved ✔️

Your project is complete!  There is nothing else to say other than... it's time to merge it  🚢 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request closed without merging**